modules/local/busco-auto-lineage.nf

```diff
@@ -0,0 +1,107 @@
+process BUSCO {
+    tag "$meta.id"
+    label 'process_large'
+
+    conda (params.enable_conda ? "bioconda::busco=5.4.3" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/busco:5.4.3--pyhdfd78af_0':
+        'quay.io/biocontainers/busco:5.4.3--pyhdfd78af_0' }"
+        
+    publishDir(
+        path: "${params.outdir}/BUSCO",
+        mode: 'copy',
+        saveAs: { fn -> fn.substring(fn.lastIndexOf('/')+1) },
+    )
+
+    input:
+    tuple val(meta), path('tmp_input/*')    // path('tmp_input/*')
+    val lineage_scale                       // Taxonomically shallow- or broad-scale busco database to compare against?
+    path busco_lineages_path                // Recommended: path to busco lineages - downloads if not set
+    //val lineage                           // Required:    lineage to check against, "auto" enables --auto-lineage instead
+    //path config_file                      // Optional:    busco configuration file
+
+    output:
+    tuple val(meta), path("*-busco.batch_summary.txt"), emit: batch_summary
+    tuple val(meta), path("short_summary.*.txt")      , emit: short_summaries_txt, optional: true
+    tuple val(meta), path("short_summary.*.json")     , emit: short_summaries_json, optional: true
+    tuple val(meta), path("*-busco")                  , emit: busco_dir
+    path "versions.yml"                               , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    // always gets set as the file itself, excluding the path
+    script:
+
+    // All I want is to be able to save the path leading to it as a variable that I can use in the busco script below. 
+    def args = task.ext.args ?: ''
+    def busco_mode = "--mode ${meta.mode}"
+    //def prefix = task.ext.prefix ?: "${meta.id}"
+    def busco_lineage_dir = busco_lineages_path ? "--offline --download_path ${busco_lineages_path}" : ''
+    def busco_lineage_scale = lineage_scale.equals('taxon_specific') ? "${meta.tax1}" : "${meta.tax2}"
+    def prefix = lineage_scale.equals('taxon_specific') ? "${meta.id}-shallow" : "${meta.id}-euk"
+    //def busco_lineage = lineage.equals('auto') ? '--auto-lineage' : "--lineage_dataset ${lineage}"
+    //def busco_config = config_file ? "--config $config_file" : ''
+    //def mode = 'proteins'
+    """
+    # Nextflow changes the container --entrypoint to /bin/bash (container default entrypoint: /usr/local/env-execute)
+    # Check for container variable initialisation script and source it.
+    if [ -f "/usr/local/env-activate.sh" ]; then
+        set +u  # Otherwise, errors out because of various unbound variables
+        . "/usr/local/env-activate.sh"
```

Comment:
Sorry, I should probably delete the auto-lineage version since I'm no longer using it - I just didn't want to lose it since I made a bunch of modifications to make it work. 

But, this section that you've highlighted and all of these checks are included in the nf-core busco module, so again, in the spirit of not fixing what isn't broken, I left it all since it didn't seem to introduce any problems.